tests/test_official.py

```diff
@@ -123,6 +123,23 @@ def check_method(h4):
 
     assert (sig.parameters.keys() ^ checked) - ignored == set()
 
+    check_extra_params_are_keywordonly(method.__name__, checked, name)
+
+
+def check_extra_params_are_keywordonly(method_name, api_params, name_in_api):
+    for c in [telegram.Bot, telegram.Chat, telegram.Message, telegram.User]:
```

Comment:
a bit futile to test anything other than Bot since they mostly have different names anyway. We would need the test [hinrich](https://github.com/python-telegram-bot/python-telegram-bot/pull/3035#pullrequestreview-973159961) mentioned for the shortcut methods :)

Reply:
ok, moved shortcut methods to a separate test `test_official.test_shortcut_methods__extra_params_are_keyword_only`. 
commit in a few minutes

Reply:
no, not quite. To test that shortcut methods behavo correctly, we don't need to change anything in `test_official`. It suffices to adapt [this function](https://github.com/python-telegram-bot/python-telegram-bot/blob/298c5fab3b8655ebae4416341335fd9c05f2ce5a/tests/conftest.py#L471-L530) which is used in all the tests of shortcut methods. Please adapt that function such that it checks that the `additional_kwargs` are keyword-only arguments of the `shortcut`. 

The overall ideas is:

* `test_official` ensures that the positional arguments of bot methods correspond to the offical bot API and that any additional arguments are keyword-only
* `check_shortcut_signature` ensures that the signatures of the shortcut methods match the signatures of the bot methods. because the bot methods are already checked agains `test_official`, this also ensures that the shortcut methods match the official signatures

Reply:
BTW after removing the loop over the different classes, this method can be simplified by just adding the logic of the function to the body of check_method :)

Reply:
I see, changing it. question: I did a rebase master locally and the hashes changed, does it matter to you if i'll use `git push -f` or to stash and return to the previous flow?

Reply:
`additional_kwargs` doesn't include all the timeout args, filename, etc..
so i'm also checking that kw args in bot_method_sig.parameters are kw args in shortcut. 

you definitely have too many shortcut methods.. wasn't there a more maintainable way to define those?

Reply:
> `additional_kwargs` doesn't include all the timeout args, filename, etc.. so i'm also checking that kw args in bot_method_sig.parameters are kw args in shortcut.

you're right, thanks 👍 

> you definitely have too many shortcut methods.. wasn't there a more maintainable way to define those?

I do think that these shortcuts are one of the strengths of PTB 🤓